full-service/src/service/address.rs

```diff
@@ -189,6 +192,33 @@ where
             }
         }
     }
+
+    fn assign_missing_reserved_subaddresses_for_accounts(&self) -> Result<(), AddressServiceError> {
+        let conn = self.wallet_db.get_conn()?;
+        let accounts = Account::list_all(&conn)?;
+
+        for account in accounts.iter() {
+            if AssignedSubaddress::get_for_account_by_index(
+                &account.account_id_hex,
+                CHANGE_SUBADDRESS_INDEX as i64,
+                &conn,
+            )
+            .is_err()
+            {
+                let account_key: AccountKey = mc_util_serial::decode(&account.account_key).unwrap();
+                AssignedSubaddress::create(
+                    &account_key,
+                    None,
+                    CHANGE_SUBADDRESS_INDEX,
```

Comment:
Does this need to have the same `fog_enabled` check?

https://github.com/mobilecoinofficial/full-service/blob/develop/full-service/src/db/account.rs#L283

Reply:
Side note - this is fragile, referencing magic number `1`:

We probably want to have this be `DEFAULT_SUBADDRESS_INDEX + 1`

Reply:
Fog enabled accounts use the reserved subaddresses as well

Reply:
doesn't need the fog enabled check, since the mc library handles change txos. These subaddress still needs to be assigned on fog accounts to pick up the txos

Reply:
Ok, I think I understand - the reason we don't need the fog enabled check is because on fog all change goes to subaddress 0 anyway?

Reply:
no they go to the change subaddress defined in the mc library, but the PublicAddress is built in such a way that the fog hint uses subaddress 0's info so that it appears in a query to Fog for txos, but lives on the correct change subaddress index

Reply:
https://github.com/mobilecoinfoundation/mobilecoin/blob/302bf9f0c238c86ec517d1e75308c68f8e5135d8/transaction/std/src/transaction_builder.rs#L312

Reply:
https://github.com/mobilecoinfoundation/mobilecoin/blob/302bf9f0c238c86ec517d1e75308c68f8e5135d8/transaction/std/src/transaction_builder.rs#L328

Reply:
Great, ty!